### PR TITLE
All Function from enumerable complete

### PR DESCRIPTION
**Pull request by konyan**

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+module Enumerable #:nodoc:
+  def my_each
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
```
Comment (stratospherique) [inline main.rb]: - [ ] my_each should return an **Enumerator** if no block is given. 
- Hint: Use `if block_given?` at the beginning of the method :wink:

File: main.rb [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    new_array
+  end
+
+  def my_all?
+    return true if size < 1
+    return false if block_given? && nil? || !block_given?
+
+    my_each { |x| return false unless yield(x) }
+    true
+  end
```
Comment (stratospherique) [inline main.rb]: * [ ] my_all? when no block or argument is given returns true when none of the collection members are false or nil
  ```ruby
    array = [1,2,5,4,7]
    p array.my_all? == array.all? # false
  ```

* [ ] my_all? should be able to accept **literals, classes i.e Integer), regex expressions** as argument. Yours crashes when doing so :crying_cat_face:

File: main.rb [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+  def my_any?
+    return false if size < 1
+    return true unless block_given?
+
+    my_each { |x| return true if yield(x) }
+    false
+  end
```
Comment (stratospherique) [inline main.rb]: * [ ] my_any? when no block or argument is given returns true when none of the collection members are false or nil
  ```ruby
    array = [1,2,5,4,7]
    p array.my_any? == array.any? # false
  ```

* [ ] my_any? should be able to accept **literals, classes i.e Integer), regex expressions** as argument. Yours crashes when doing so :crying_cat_face:

File: main.rb [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+      if arg.is_a?(Regexp)
+        my_each { |x| return false if pattern =~ x.to_s }
+      else
+        my_each { |x| return false if x.is_a?(pattern) }
+      end
+    else
+      my_each { |x| return false if x }
+    end
+    true
+  end
```
Comment (stratospherique) [inline main.rb]: * [ ] my_none? should be able to accept **literals, classes i.e Integer), regex expressions** as argument. Yours crashes when doing so :crying_cat_face:

File: main.rb [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    arr
+  end
+
+  def my_inject(*args)
+    init = args.size.positive?
+    acc = init ? args[0] : self[0]
+    drop(init ? 0 : 1).my_each { |item| acc = yield(acc, item) }
+    acc
+  end
+end
```
Comment (stratospherique) [inline main.rb]: * [ ] my_inject when a symbol is specified combines each element of the collection by applying the symbol as a named method
  ```
    array = [1,5,8,7,9,5,5,55,5]
    p array.my_inject(:+) 
    # return this errors 
    LocalJumpError:
        no block given (yield)
     # ./main.rb:92:in `block in my_inject'
      # ./main.rb:7:in `my_each'
      # ./main.rb:92:in `my_inject'
  ```

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great Start @konyan :clap: :clap: However, some requirements needs to be implemented in order for us to approve your project. cehckt :arrow_down: 

- **[REQUIRED]** Check code comments below. they should help you refactor your methods :nerd_face: 

### That's it :relaxed: When finished, kindly resubmit another code review request :pray: 

Happy coding :v: 
**Ahmed Mahfoudh**

**Comment by shshamim63** [comment]
## Status :recycle: [!! Required Changes !!] :recycle: 
___
Hi @konyan  :wave: ,
Awesome job :clap: . You are almost there. First of all you have covered most of the task mentioned by the previous reviewer. But i have noticed some edge cases which are actually making the methods acting like the original one. And I am sure that you will cover them easily.
___
##### Please follow the instructions below :point_down: to complete this project :smile: 
___
### :spaghetti: [Mandatory changes] :spaghetti: 
___
1. **my_all?**  If instead a pattern is supplied, the method returns whether pattern === element for every collection member. 
```
[3, 3, 3].my_all?(3)
```
should return true.
image
2. **my_none?** Is the total opposite of **my_any?**
https://github.com/konyan/enumerable-methods/blob/b5b794da1310a75c47650af46a0a114b0d8eeb8c/main.rb#L68-L71
From where the **pattern** is coming. I think it should be arg.
If I run `[1, 2, 3].my_none?(3)` it should return false.
3. **my_inject**  it can take two arguments (initial, symbol). If I execute
```
[1, 2, 3].my_inject(2){ |sum, n| sum + n }
[1, 2, 3].my_inject(2, :+)
```
both of them should provide 8.
image
https://github.com/konyan/enumerable-methods/blob/b5b794da1310a75c47650af46a0a114b0d8eeb8c/main.rb#L118-L124
What you tried to do :point_up: I could not get it. There is a simple way to do that. Check out [this](https://stackoverflow.com/questions/3337285/what-does-send-do-in-ruby) link. You will get an idea how to cover it.
___
#### Please fix the above mentioned issues and send another review request.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain
[Linkedin](https://www.linkedin.com/in/shakhawathossainshamim/) | [Slack](Shakhawat Hossain)

**Comment by shshamim63** [comment]
## Status :green_apple: [!! Approved !!] :green_apple: 
___
Hi @konyan :wave: , 
awesome job :clap: . Most of the methods are working properly. Just found a funny cause that is actually blocking my_none to check some test cases. And the issue is really simple. So that is why I am marking this project as approved in advance. But you have to make a small change. :fire:  [PR should contain a short description about the task you have covered in here].
___
## :spades: [Mandatory Changes] :spades: 
___
1. I couldn't find the argument pattern. I believe it should be arg. Please replace pattern with arg.
https://github.com/konyan/enumerable-methods/blob/850c07da6e37b990b2d37ffdec0675cd7668312c/main.rb#L71-L73
___
#### Please fix the above issue and after that you are good to go.
___
## :gear: Happy Coding :gear: 
### Shakhawat Hossain
[Linkedin](https://www.linkedin.com/in/shakhawathossainshamim/) | [Slack](@Shakhawat)

**Pull request merged**